ddtrace/contrib/internal/openai/utils.py

```diff
@@ -162,6 +162,77 @@ async def _extract_token_chunk(self, chunk):
             return
 
 
+class TracedOpenAIResponseStream(BaseTracedOpenAIStream):
+    def __enter__(self):
+        self.__wrapped__.__enter__()
+        return self
+
+    def __exit__(self, exc_type, exc_val, exc_tb):
+        self.__wrapped__.__exit__(exc_type, exc_val, exc_tb)
+
+    def __iter__(self):
+        exception_raised = False
+        try:
+            for chunk in self.__wrapped__:
+                if chunk.type == "response.completed":
+                    handle_response_tools(chunk, self._dd_span)
+                    self._dd_integration.record_usage(self._dd_span, chunk.usage)
+                    self._dd_span.finish()
+                yield chunk
+        except Exception:
+            self._dd_span.set_exc_info(*sys.exc_info())
+            exception_raised = True
+            raise
+        finally:
+            if not exception_raised:
+                _process_finished_stream(
+                    self._dd_integration, self._dd_span, self._kwargs, self._streamed_chunks, self._is_completion
+                )
+            self._dd_span.finish()
+
+    def __next__(self):
+        try:
+            chunk = self.__wrapped__.__next__()
+            _loop_handler(self._dd_span, chunk, self._streamed_chunks)
+            return chunk
+        except StopIteration:
+            _process_finished_stream(
+                self._dd_integration, self._dd_span, self._kwargs, self._streamed_chunks, self._is_completion
+            )
+            self._dd_span.finish()
+            raise
+        except Exception:
+            self._dd_span.set_exc_info(*sys.exc_info())
+            self._dd_span.finish()
+            raise
+
+
+def handle_response_tools(resp, span):
+    """Process and set tool information from the response to the span.
+
+    Args:
+        resp: The response object containing tool information
+        span: The span to set the tool information on
+    """
+    if not getattr(resp, "tools"):
```

Comment:
```suggestion
    if not hasattr(resp, "tools"):
```
i would do this to be safe - if for some reason the object is a pydantic object, it'll throw errors instead of returning `None`